frontend/www/js/omegaup/arena/events_socket.ts

```diff
@@ -0,0 +1,218 @@
+import * as time from '../time';
+import * as ui from '../ui';
+import * as api from '../api';
+import {
+  ContestClarificationType,
+  refreshContestClarifications,
+} from './clarifications';
+import clarificationStore from './clarificationsStore';
+import { onRankingChanged, onRankingEvents } from './ranking';
+import { updateRun } from './submissions';
+
+export enum SocketStatus {
+  Waiting = '↻',
+  Failed = '✗',
+  Ok = '•',
+}
+
+export class EventsSocket {
+  uri: string = '';
+  shouldRetry: boolean = false;
+  socket: WebSocket | null = null;
+  socketKeepalive: ReturnType<typeof setTimeout> | null = null;
+  retries: number = 10;
+
+  disableSockets: boolean = false;
+  problemsetAlias: string | null = null;
+  locationProtocol: null | string = null;
+  locationHost: null | string = null;
+  problemsetId: null | number = null;
+  scoreboardToken: null | string = null;
+  socketStatus: SocketStatus = SocketStatus.Waiting;
+  clarificationInterval: ReturnType<typeof setTimeout> | null = null;
+  rankingInterval: ReturnType<typeof setTimeout> | null = null;
+  clarificationsOffset: number = 0;
+  clarificationsRowcount: number = 20;
+
+  constructor(options?: {
+    disableSockets: boolean;
+    problemsetAlias: string;
+    locationProtocol: string;
+    locationHost: string;
+    problemsetId: number;
+    scoreboardToken: string;
+    clarificationsOffset: number;
+    clarificationsRowcount: number;
+  }) {
```

Comment:
aquí se prefiere el uso de parámetros default:

```suggestion
  constructor({
      disableSockets = false,
      problemsetAlias,
      locationProtocol,
      locationHost,
      problemsetId,
      scoreboardToken = null,
      clarificationsOffset = 0,
      clarificationsRowcount = 20,
    }: {
      disableSockets: boolean;
      problemsetAlias: string;
      locationProtocol: string;
      locationHost: string;
      problemsetId: number;
      scoreboardToken: string;
      clarificationsOffset: number;
      clarificationsRowcount: number;
    } = {}
  ) {
```

así todas las propiedades en L25-L35 se pueden declarar sin ser nulables! además, así no se requiere que las cosas que tienen default se declaren explícitamente si sólo se quieren pasar algunos parámetros y además así no es posible pasarle un conjunto inválido de parámetros: un `locationHost` nulo es ilegal porque una instancia con eso nulo no funcionaría!

Reply:
Listo, le tuve que quitar el `= {}`, porque me estaba marcando error de que faltaban las propiedades que no tienen valor default